_posts/2020-01-01-Resurrecting-Sigmoid.markdown

```diff
@@ -0,0 +1,521 @@
+- Poole, Lhiri, Raghu, Sohl-Dickstein & Ganguli: [Exponential expressivity in deep neural networks through transient chaos
+](https://arxiv.org/abs/1606.05340) (Sections 1, 2, and 3)
+- Schoenholz, Gilmer, Ganguli, & Sohl-Dickstein: [Deep information propagation](https://arxiv.org/pdf/1611.01232.pdf) (Sections 1, 2, 3, and 5)
+
+These are very useful references, but for those unfamiliar with the field, not necessarily pedagogical.  The problem set listed below is designed to walk you through understanding the formalism in a self-contained manner.  We strongly suggest working through the problem set before reading the papers above, and only consulting afterwards or for reference.  Certain problems in the problem set point to sections of the above papers for hints.  
+
+**Optional Reading**:
+
+Once you understand the mean-field analysis framework, you will have a good foundation for understanding the following papers.  These are strictly 'bonus', i.e. not connected to the main paper of this course.
+1. [Deep Neural Networks as Gaussian Processes](https://arxiv.org/abs/1711.00165)
+2. [Wide Neural Networks of Any Depth Evolve as Linear Models under Gradient Descent](https://arxiv.org/abs/1902.06720)d
+
+**Questions**:
+
+The full problem set, from which the below problems are taken, is [here](/assets/sigmoid/problem-sets/pdfs/1.pdf).
+
+1. Problem 2 from full set: The mean field approximation
+
+In this problem, we use the knowledge we gained in problem 1 to properly choose to initialize the weights and biases according to $$W^l \sim \mathcal{N}(0, \sigma_w^2/N)$$ and $$b^l \sim \mathcal{N}(0, \sigma_b^2)$$. We'll investigate some techniques that will be useful in understanding precisely how the network's random initialization influences what the net does to its inputs; specifically, we'll be able to take a look at how the \textit{depth} of the network together with the initialization governs the propagation of an input point as it flows forward through the network's layers.
+
+a. A natural property of input points to study as the input flows through the net layer by layer is its length. Intuitively, this is closely related to how the net transforms the input space, and to how the depth of the net relates to that transformation. Compute the length $$q^l$$ of the activation vector outputted by layer $$l$$. When considering non-rectangular nets, where layer $$l$$ has length $$N_l$$, we want to distinguish this activation norm from the width of individual layers, so what's a more appropriate quantity we can track to understand how the lengths of activation vectors change in the net?
+
+b. What probabilistic quantity of the neuronal activations does $$q^l$$ approximate (with the approximation improving for larger $$N$$)?
+_Hint: recall that all neuronal activations $$h^l_i$$ are zero-mean, and consider the definition of $$q^l$$ from part (a) in terms of the empirical distribution of $$h^l_i$$.
```

Comment:
Missing closing underscore at the end of the hint; this is why the rest of the problems below are all italicized.